### ISSUE TITLE
[debops.dovecot] username stripping

### ISSUE THREAD

**Issue by sread** [debops/debops#523]
Hi,

I changed my dovecot configuration from LDA to LMTP, and I found that for remote mail sent via postfix->LMTP->dovecot to a system user, I had to add

    auth_username_format = %Ln

to the `/etc/dovecot/conf.d/10-auth.conf` file or Dovecot would reject the original user@domain.com as "User doesn't exist".

Is this a hidden configuration problem I'm missing? Or something missing from debops.

Thanks!

**Comment by drybjed** [comment]
It's most likely something missing from DebOps. Do you want to add it conditionally, when LMTP is enabled?

**Comment by sread** [comment]
Yes, and probably only when the "system" `dovecot_user_accounts` method is active?

My concern is that for people using SQL or virtual mailboxes, this is too heavy of a hammer, particularly if they rely on being passed both the username and domain.

- https://serverfault.com/questions/585362/virtual-and-system-mailboxes-how-to-set-passdb-propperly?rq=1
 - https://wiki2.dovecot.org/Authentication/MultipleDatabases

**Comment by drybjed** [comment]
Sure, sounds good for a start. I think that entire role should rewritten in the `debops.freeradius` style with some default configuration, so that different parts of the config can be activated conditionally. Perhaps even split into multiple roles that manage IMAP/POP3, with maybe some central role that works a'la `debops.postfix` and manages the config files. At least, I see this in that role's future.
